### ISSUE TITLE
Current select template does not support ngOption's 'track by value' setting

### ISSUE THREAD

**Issue by rmulder** [formly-js/angular-formly-templates-bootstrap#23]
Please review the change I made to the file: src/types/select.html in my fork of this repo at the following commit: https://github.com/rmulder/angular-formly-templates-bootstrap/commit/d7dcc6bf024d0850e49d2dba8eca38410dd88d37#diff-e34487d2e565168713f0e02aaf3f89cc

I was pulling my hair out today trying to figure out why the 'value' attribute of the option tag was not filling in the values I was setting it to - until I realized that this select template does not support the current ng-options syntax. Please update this template at your earliest convenience, then I can change our bower.json back to point to the 'standard' version here...

Thanks!

**Comment by kentcdodds** [comment]
Thanks for that @rmulder. I'll try to get it updated as soon as I'm able. Actually, my select type at work is a lot more sophisticated. I think I may just use that one. Here it is in case you're interested:

```javascript
export default ngModule => {
  const template =
    `<select ng-model="model[options.key]" class="form-control">
      <option ng-if="!to.notNull" value="">{{to.nullDisplay}}</option>
    </select>`;
  ngModule.run((formlyConfig) => {
    formlyConfig.setType({
      name: 'select',
      extends: 'azBase',
      template,
      defaultOptions(options) {
        const forWithFilter = `for item in to.options | azDynamicFilter:to.filter:to.orderBy`;
        let ngOptions = `item[to.valueProp] as item[to.displayProp || 'name'] ${forWithFilter}`;
        const {valueProp, displayProp, options: items} = options.templateOptions;
        if (items && typeof items[0] === 'string') {
          ngOptions = `item for item in to.options | azDynamicFilter:to.filter:'toString()'`;
        } else if (valueProp === null && displayProp) {
          ngOptions = `item as item[to.displayProp] ${forWithFilter}`;
        }
        return {
          ngModelAttrs: {
            [ngOptions]: {
              value: 'ng-options'
            }
          },
          templateOptions: {
            filter: 'orderBy',
            orderBy: 'name',
            displayProp: 'name',
            valueProp: 'value'
          }
        };
      }
    });
  });
};
```

Note, this uses something called the `azDynamicFilter` which is pretty cool too :-)

```javascript
export default ngModule => {
  ngModule.filter('azDynamicFilter', function($filter) {
    return function azDynamicFilter(input, filterName) {
      if (!filterName) {
        return input;
      }
      // filterName can contain its arguments for convenience
      var args = [];
      getFilterArgs(filterName, args);
      if (args.length === 1) {
        args = Array.prototype.slice.call(arguments, 0);
        args.splice(1, 1); // remove filter name
      } else {
        filterName = args[0];
        args.splice(0, 1, input); // remove the filter name
      }
      var filter = $filter(filterName);
      return filter.apply(null, args); // invoke filter with args received

      function getFilterArgs(str, matches) {
        /* eslint complexity:[2, 7] */
        if (!str) {
          return matches;
        }
        var startIndex = 0;
        var nextColon = str.indexOf(':');
        var endIndex = str.length - (nextColon !== -1 ? 1 : 0);
        var isQuoted = str.indexOf('"') === 0;
        if (isQuoted) {
          startIndex = 1;
          endIndex = str.substr(1).indexOf('"') + 1;
        } else if (nextColon > 0) {
          endIndex = nextColon;
        }
        var head = str.substring(startIndex, endIndex).trim();
        matches.push(head);
        var tail = str.substring(head.length + (isQuoted ? 2 : 1));
        if (tail.indexOf(':') === 0) {
          tail = tail.substr(1);
        }
        return getFilterArgs(tail, matches);
      }
    };
  });
};
```

It allows you to use a filter with a dynamic name.

**Comment by kentcdodds** [comment]
After working on it [for a while](https://github.com/formly-js/angular-formly-templates-bootstrap/tree/feature/improve-select) I decided that an easier way to go about it is to simply add a `templateOptions.ngOptions` property. Now you can simply pass your own in. Hope this helps! This is now available in 4.1.0!
